pkg/test/framework/components/istio/operator.go

```diff
@@ -0,0 +1,183 @@
+	scopes.CI.Infof("=== BEGIN: Cleanup Istio ===")
+	defer scopes.CI.Infof("=== DONE: Cleanup Istio ===")
+	if i.settings.DeployIstio {
+		err = i.environment.Accessor.DeleteNamespace(i.settings.SystemNamespace)
+		if err == nil {
+			err = i.environment.Accessor.WaitForNamespaceDeletion(i.settings.SystemNamespace)
+		}
+		// Note: when cleaning up an Istio deployment, ValidatingWebhookConfiguration
+		// and MutatingWebhookConfiguration must be cleaned up. Otherwise, next
+		// Istio deployment in the cluster will be impacted, causing flaky test results.
+		// Clean up ValidatingWebhookConfiguration, if any
+		_ = i.environment.DeleteValidatingWebhook(DefaultValidatingWebhookConfigurationName)
+		// Clean up MutatingWebhookConfiguration, if any
+		_ = i.environment.DeleteMutatingWebhook(DefaultMutatingWebhookConfigurationName)
+	}
+	return
+}
+
+func (i *operatorComponent) Dump() {
+	scopes.CI.Errorf("=== Dumping Istio Deployment State...")
+
+	d, err := i.ctx.CreateTmpDirectory("istio-state")
+	if err != nil {
+		scopes.CI.Errorf("Unable to create directory for dumping Istio contents: %v", err)
+		return
+	}
+
+	deployment.DumpPodState(d, i.settings.SystemNamespace, i.environment.Accessor)
+	deployment.DumpPodEvents(d, i.settings.SystemNamespace, i.environment.Accessor)
+
+	pods, err := i.environment.Accessor.GetPods(i.settings.SystemNamespace)
+	if err != nil {
+		scopes.CI.Errorf("Unable to get pods from the system namespace: %v", err)
+		return
+	}
+
+	for _, pod := range pods {
+		for _, container := range pod.Spec.Containers {
+			l, err := i.environment.Logs(pod.Namespace, pod.Name, container.Name, false /* previousLog */)
+			if err != nil {
+				scopes.CI.Errorf("Unable to get logs for pod/container: %s/%s/%s", pod.Namespace, pod.Name, container.Name)
+				continue
+			}
+
+			fname := path.Join(d, fmt.Sprintf("%s-%s.log", pod.Name, container.Name))
+			if err = ioutil.WriteFile(fname, []byte(l), os.ModePerm); err != nil {
+				scopes.CI.Errorf("Unable to write logs for pod/container: %s/%s/%s", pod.Namespace, pod.Name, container.Name)
+			}
+		}
+	}
+}
+
+func deployOperator(ctx resource.Context, env *kube.Environment, cfg Config) (Instance, error) {
+	scopes.CI.Infof("=== Istio Component Config ===")
+	scopes.CI.Infof("\n%s", cfg.String())
+	scopes.CI.Infof("================================")
+
+	i := &operatorComponent{
+		environment: env,
+		settings:    cfg,
+		ctx:         ctx,
+	}
+	i.id = ctx.TrackResource(i)
+
+	if !cfg.DeployIstio {
+		scopes.Framework.Info("skipping deployment due to Config")
+		return i, nil
+	}
+
+	istioCtl, err := istioctl.New(ctx, istioctl.Config{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Top-level work dir for Istio deployment.
+	workDir, err := ctx.CreateTmpDirectory("istio-deployment")
+	if err != nil {
+		return nil, err
+	}
+
+	icpFile := filepath.Join(workDir, "icp.yaml")
+	if err := ioutil.WriteFile(icpFile, []byte(cfg.IstioControlPlane()), os.ModePerm); err != nil {
```

Comment:
any concern about `IstioControlPlane` silently failing here (and returning "")?

Reply:
Good point, I think this is actually normally returning `""` and then that does nothing since `-f ""` is a no-op. So we should probably clean up this a bit